dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelperTest.java

```diff
@@ -0,0 +1,349 @@
+                    .get(parentContentType.variable().toLowerCase())).get("properties");
+
+            //validate age mapping results
+            final Map ageMapping = ((JSONObject) contentTypeJSON
+                    .get(ageField.variable().toLowerCase())).getAsMap();
+            assertNotNull(ageMapping);
+            assertEquals(1, ageMapping.size());
+            assertEquals("long", ageMapping.get("type"));
+
+            //validate relationship field mapping results
+            final Map relationshipMapping = ((JSONObject) contentTypeJSON
+                    .get(relationshipField.variable().toLowerCase())).getAsMap();
+            assertNotNull(relationshipMapping);
+            assertEquals(1, relationshipMapping.size());
+            assertEquals("text", relationshipMapping.get("type"));
+
+            //validate legacy relationship
+            final Map legacyRelationshipMapping = ((JSONObject) propertiesJSON
+                    .get(legacyRelationship.getRelationTypeValue().toLowerCase())).getAsMap();
+            assertNotNull(ageMapping);
+            assertEquals(2, legacyRelationshipMapping.size());
+            assertEquals("keyword", legacyRelationshipMapping.get("type"));
+            assertEquals(8191, legacyRelationshipMapping.get("ignore_above"));
+
+        } finally {
+            if (workingIndex != null) {
+                contentletIndexAPI.delete(workingIndex);
+            }
+
+            if (parentContentType != null && parentContentType.inode() != null) {
+                ContentTypeDataGen.remove(parentContentType);
+            }
+
+            if (childContentType != null && childContentType.inode() != null) {
+                ContentTypeDataGen.remove(childContentType);
+            }
+        }
+    }
+
+    /**
+     * Creates a new Legacy Relationship object
+     * @param parentContentType
+     * @param childContentType
+     * @return
+     */
+    private Relationship createLegacyRelationship(final ContentType parentContentType,
+            final ContentType childContentType) {
+        final String relationTypeValue = parentContentType.name() + "-" + childContentType.name();
+
+        Relationship relationship;
+        relationship = relationshipAPI.byTypeValue(relationTypeValue);
+        if (null != relationship) {
+            return relationship;
+        } else {
+            relationship = new Relationship();
+            relationship.setParentRelationName(parentContentType.name());
+            relationship.setChildRelationName(childContentType.name());
+            relationship.setCardinality(0);
+            relationship.setRelationTypeValue(relationTypeValue);
+            relationship.setParentStructureInode(parentContentType.inode());
+            relationship.setChildStructureInode(childContentType.id());
+            try {
+                relationshipAPI.create(relationship);
+            } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)